src/main/java/org/broadinstitute/hellbender/tools/AnalyzeSaturationMutagenesis.java

```diff
@@ -1614,21 +1610,81 @@ public void updateCounts( final MoleculeCounts moleculeCounts,
         return new Interval(readStart, readEnd);
     }
 
-    @VisibleForTesting static void updateCountsForPair( final ReadReport report1, final ReadReport report2 ) {
+    private static Interval calculateShortFragmentTrim( final GATKRead read, final Interval qualityTrim ) {
+        if ( qualityTrim.size() < minLength ) return qualityTrim;
+
+        // don't process past end of fragment when fragment length < read length
+        if ( read.isProperlyPaired() ) {
+            final int fragmentLength = Math.abs(read.getFragmentLength());
+            if ( read.isReverseStrand() ) { // the read has been RC'd: trim its beginning if necessary
+                final int minStart = read.getLength() - fragmentLength;
+                if ( qualityTrim.getStart() < minStart ) {
+                    if ( qualityTrim.getEnd() - minStart < minLength ) { // if there are too few calls left
+                        return Interval.NULL_INTERVAL;
+                    }
+                    return new Interval(minStart, qualityTrim.getEnd());
+                }
+            } else if ( fragmentLength < qualityTrim.getEnd() ) { // trim the end of the read, if necessary
+                if ( fragmentLength - qualityTrim.getStart() < minLength ) { // if there are too few calls left
+                    return Interval.NULL_INTERVAL;
+                }
+                return new Interval(qualityTrim.getStart(), fragmentLength);
+            }
+        }
+        return qualityTrim;
+    }
+
+    @VisibleForTesting static void updateCountsForPair( final GATKRead read1, final ReadReport report1,
+                                                        final GATKRead read2, final ReadReport report2 ) {
         if ( report1.getRefCoverage().isEmpty() ) {
-            report2.updateCounts(moleculeCountsDisjointPair, codonTracker, variationCounts, reference);
+            if ( !report2.getRefCoverage().isEmpty() ) {
+                processReport(read2, report2, disjointPairCounts);
+            }
         } else if ( report2.getRefCoverage().isEmpty() ) {
-            report1.updateCounts(moleculeCountsDisjointPair, codonTracker, variationCounts, reference);
+            if ( !report1.getRefCoverage().isEmpty() ) {
```

Comment:
IDE tells me `report1.getRefCoverage().isEmpty()` is always false, which seem to be write given that the enclosing block is triggered only when `report1.getRefCoverage().isEmpty()` is false.

But I can understand if it is for symmetry.

Reply:
Ha.  Just silliness.  I removed the redundant test.